cmd/secrets/secret_helper.go

```diff
@@ -19,32 +19,62 @@ import (
 
 	"github.com/spf13/cobra"
 
+	"github.com/DataDog/datadog-agent/cmd/secrets/providers"
 	s "github.com/DataDog/datadog-agent/pkg/secrets"
 )
 
+// This executable provides a "read" command to fetch secrets. It can be used in
+// 2 different ways:
+//
+// 1) With the "--with-provider-prefixes" option enabled. Each input secret
+// should follow this format: "providerPrefix/some/path". The provider prefix
+// indicates where to fetch the secrets from. At the moment, we support "file"
+// and "k8s_secret". The path can mean different things depending on the
+// provider. In "file" it's a file system path. In "k8s_secret", it follows this
+// format: "namespace/name/key".
+//
+// 2) Without the "--with-provider-prefixes" option. The program expects a root
+// path in the arguments and input secrets are just paths relative to the root
+// one. So for example, if the secret is "my_secret" and the root path is
+// "/some/path", the fetched value of the secret will be the contents of
+// "/some/path/my_secret". This option was offered before introducing
+// "--with-provider-prefixes" and is kept to avoid breaking compatibility.
+
 const (
-	maxSecretFileSize = 8192
+	providerPrefixesFlag    = "with-provider-prefixes"
+	providerPrefixSeparator = "@"
 )
 
 func init() {
-	SecretHelperCmd.AddCommand(readSecretCmd)
+	cmd := readSecretCmd
+	cmd.Flags().Bool(providerPrefixesFlag, false, "Use prefixes to select the secrets provider (file, k8s_secret)")
+	SecretHelperCmd.AddCommand(cmd)
 }
 
-// SecretHelperCmd implements secrets backend helper commands
+// SecretHelperCmd implements secrets provider helper commands
 var SecretHelperCmd = &cobra.Command{
 	Use:   "secret-helper",
-	Short: "Secret management backend helper",
+	Short: "Secret management provider helper",
 	Long:  ``,
 }
 
-// ReadSecretsCmd implements reading secrets from a directory/volume mount
 var readSecretCmd = &cobra.Command{
 	Use:   "read",
-	Short: "Read secret from a directory",
+	Short: "Read secrets",
 	Long:  ``,
-	Args:  cobra.ExactArgs(1),
+	Args:  cobra.MaximumNArgs(1), // 0 when using the provider prefixes option, 1 when reading a file
 	RunE: func(cmd *cobra.Command, args []string) error {
-		return readSecrets(os.Stdin, os.Stdout, args[0])
+		usePrefixes, err := cmd.Flags().GetBool(providerPrefixesFlag)
+		if err != nil {
+			return err
+		}
+
+		dir := ""
+		if len(args) == 1 {
+			dir = args[0]
+		}
+
+		return readSecrets(os.Stdin, os.Stdout, dir, usePrefixes, &providers.Selector{})
```

Comment:
The selector is only use in `readSecretsUsingPrefixes`, why not create it in that function ?